### PR TITLE
CppCheck DemandManager and DesiccantDehumidifiers

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck style issues
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DemandManager.cc [hunk 1]
```diff
@@ -694,7 +660,7 @@ void GetDemandManagerInput(EnergyPlusData &state)
             DemandMgr(MgrNum).Type = DemandMgr(MgrNum).Type = ManagerType::ExtLights;
```
Comment (amirroth) [inline src/EnergyPlus/DemandManager.cc]: Not doing the `auto &demandMgr` thing in this function?

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 2]
```diff
@@ -232,48 +212,13 @@ namespace DesiccantDehumidifiers {
...
         // Obtains input data for humidifiers and stores it in dehumidifier data structures.
 
         // METHODOLOGY EMPLOYED:
         // Uses InputProcessor "Get" routines to obtain data.
 
-        // Using/Aliasing
-        using BranchNodeConnections::SetUpCompSets;
-        using BranchNodeConnections::TestCompSet;
-        using NodeInputManager::GetOnlySingleNode;
-        auto &GetDXCoilOutletNode(DXCoils::GetCoilOutletNode);
```
Comment (amirroth) [inline src/EnergyPlus/DesiccantDehumidifiers.cc:247]: This is an obnoxious use of `auto` that I am trying to get rid of (among the other ones).

File: src/EnergyPlus/DemandManager.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/DemandManager.cc]: I probably missed that one. I intended to do them all.

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 3]
```diff
@@ -374,7 +309,7 @@ namespace DesiccantDehumidifiers {
             if (lAlphaBlanks(2)) {
```
Comment (amirroth) [inline src/EnergyPlus/DesiccantDehumidifiers.cc:309]: Maybe do the `auto &desicDehum` thing here.

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 4]
```diff
@@ -2544,81 +2427,72 @@ namespace DesiccantDehumidifiers {
...
-            ShowRecurringContinueErrorAtEnd(
-                state, DesicDehum(DesicDehumNum).DehumType + '=' + DesicDehum(DesicDehumNum).Name, DesicDehum(DesicDehumNum).RegenFanErrorIndex2);
+                                          desicDehum.RegenFanErrorIndex1);
+            ShowRecurringContinueErrorAtEnd(state, desicDehum.DehumType + '=' + desicDehum.Name, desicDehum.RegenFanErrorIndex2);
             RhoAirStdInit = state.dataEnvrn->StdRhoAir;
             ShowRecurringContinueErrorAtEnd(state,
-                                            "Flow requested [m3/s] from " + DesicDehum(DesicDehumNum).RegenFanType + '=' +
-                                                DesicDehum(DesicDehumNum).RegenFanName,
-                                            DesicDehum(DesicDehumNum).RegenFanErrorIndex3,
+                                            "Flow requested [m3/s] from " + desicDehum.RegenFanType + '=' + desicDehum.RegenFanName,
```
Comment (amirroth) [inline src/EnergyPlus/DesiccantDehumidifiers.cc]: `format`

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 5]
```diff
@@ -2544,81 +2427,72 @@ namespace DesiccantDehumidifiers {
...
-                DesicDehum(DesicDehumNum).RegenCapErrorIndex1);
-            ShowRecurringContinueErrorAtEnd(
-                state, DesicDehum(DesicDehumNum).DehumType + '=' + DesicDehum(DesicDehumNum).Name, DesicDehum(DesicDehumNum).RegenCapErrorIndex2);
+                desicDehum.RegenCapErrorIndex1);
+            ShowRecurringContinueErrorAtEnd(state, desicDehum.DehumType + '=' + desicDehum.Name, desicDehum.RegenCapErrorIndex2);
             ShowRecurringContinueErrorAtEnd(state,
-                                            "Load requested [W] from " + DesicDehum(DesicDehumNum).RegenCoilType + '=' +
-                                                DesicDehum(DesicDehumNum).RegenCoilName,
-                                            DesicDehum(DesicDehumNum).RegenCapErrorIndex3,
+                                            "Load requested [W] from " + desicDehum.RegenCoilType + '=' + desicDehum.RegenCoilName,
```
Comment (amirroth) [inline src/EnergyPlus/DesiccantDehumidifiers.cc]: and here

**Comment by rraustad** [comment]
@Myoldmopar I do have a branch with comments addressed. I haven 't pushed yet since I was waiting for a lull in CI. I will push this later this evening.

**Comment by Myoldmopar** [comment]
@rraustad I am now pretty much set on team contributions for 23.1, so CI will be settling down.  You are welcome to push your branch anytime you like.  Just make sure to pull develop in.  If it happens to be all super clean and no issues, I may drop it in as I'm wrapping things up.

**Comment by Myoldmopar** [comment]
I'm going to try to sneak in these 2 cppcheck branches and the final auto branch while I have other work going.  CI is all super green here, so this is an easy start.  Testing locally...

File: src/EnergyPlus/DesiccantDehumidifiers.hh [hunk 6]
```diff
@@ -305,8 +305,6 @@ struct DesiccantDehumidifiersData : BaseGlobalStruct
     bool GetInputDesiccantDehumidifier = true; // First time, input is "gotten"
     bool InitDesiccantDehumidifierOneTimeFlag = true;
     bool MySetPointCheckFlag = true; // I think this actually needs to be a vector or a member variable on the struct, not just a single bool
-    bool CalcSolidDesiccantDehumidifierMyOneTimeFlag = true; // one time flag
```
Comment (Myoldmopar) [inline src/EnergyPlus/DesiccantDehumidifiers.hh:308]: :heart:

**Review by Myoldmopar — APPROVED** [review]
Love it.  More variable localization, reference shortcuts, removed usings, all good.

**Pull request merged**